src/SFA.DAS.ApplyService.Data/ApplyRepository.cs

```diff
@@ -382,6 +395,96 @@ ELSE @sectionStatusInProgress
+                        ) ab",
+                        new
+                        {
+                            applicationStatusSubmitted = ApplicationStatus.Submitted,
+                            sequenceStatusSubmitted = ApplicationSequenceStatus.Submitted,
+                            sectionStatusInProgress = ApplicationSectionStatus.InProgress
+                        })).ToList();
+            }
+        }
+
+        public async Task<List<ApplicationSummaryItem>> GetClosedApplications()
+        {
+            using (var connection = new SqlConnection(_config.SqlConnectionString))
+            {
+                return (await connection
+                    .QueryAsync<ApplicationSummaryItem>(
+                        @"SELECT OrganisationName, ApplicationId, SequenceId,
+                            CASE WHEN SequenceId = 1 THEN 'Mid point'
+                                 WHEN SequenceId = 2 THEN 'Standard'
+                                 ELSE 'Unknown'
+                            END As ApplicationType,
+                            CASE WHEN SequenceId = 1 THEN JSON_VALUE(ApplicationData, '$.InitSubmissionClosedDate')
+		                         WHEN SequenceId = 2 THEN JSON_VALUE(ApplicationData, '$.StandardSubmissionClosedDate')
+		                         ELSE NULL
+	                        END As ClosedDate,
+                            SequenceStatus As CurrentStatus
+                        FROM (
+	                        SELECT 
+                                org.Name AS OrganisationName,
+                                seq.SequenceId AS SequenceId,
+                                appl.id AS ApplicationId,
+                                appl.ApplicationData AS ApplicationData,
+                                seq.Status As SequenceStatus
+	                        FROM Applications appl
+	                        INNER JOIN ApplicationSequences seq ON seq.ApplicationId = appl.Id
+	                        INNER JOIN ApplicationSections sec ON sec.ApplicationId = appl.Id 
+	                        INNER JOIN Organisations org ON org.Id = appl.ApplyingOrganisationId
+	                        WHERE appl.ApplicationStatus IN(@applicationStatusApproved, @applicationStatusRejected) 
+                                AND seq.Status IN (@sequenceStatusApproved, @sequenceStatusRejected)
+	                        GROUP BY seq.SequenceId, seq.Status, appl.ApplyingOrganisationId, appl.id, org.Name, appl.ApplicationData 
+                        ) ab",
+                        new
+                        {
+                            applicationStatusApproved = ApplicationStatus.Approved,
+                            applicationStatusRejected = ApplicationStatus.Rejected,
+                            sequenceStatusApproved = ApplicationSequenceStatus.Approved,
+                            sequenceStatusRejected = ApplicationSequenceStatus.Rejected,
+                        })).ToList();
+            }
+        }
+
+
```

Comment:
should these big calls be in a sproc?   Just wondered.  Might be worth raising a tech debt story to do that?

Reply:
If they get any bigger then I think we should. Currently it's a whole bunch of case statements based on sequence that is making it look huge